ProductsAnalysisWeb/Pages/MobileWidget.cshtml

```diff
@@ -8,16 +8,16 @@
         <h1>Most Expensive Phones</h1>
         <h2>Total count: @Model.Count</h2>
         @{
-            int maximum = Model.ToList().Select(x => x.Price).Max();
+            double maximum = Model.ToList().Select(x => x.Price).Max();
             var query = Model.ToList().Where(x => x.Price == maximum);
```

Comment:
It is dangerous to compare doubles... it can give you false negatives... You need to define predicate and compare with some tolerance...